File: Module2_SPA_Angular/TypescriptDemos/employee.ts
import { IEmployee } from "./IEmployee";

export class Employee implements IEmployee{
    ecode: number;
    ename: string;
    salary: number;
    deptid: number;
    
    
    constructor(ecode:number,ename:string,salary:number,deptid:number){
        this.ecode=ecode;
        this.ename=ename;
        this.salary=salary;
        this.deptid=deptid;
    }

    display(): void {
        console.log(`${this.ecode}\t${this.ename}\t${this.salary}\t${this.deptid}`)
    }
    calculateBonus():number{
        let bonus:number;
        if(this.salary>5000){
            bonus=0.1*this.salary;
        }else{
            bonus=0.2*this.salary;
        }

        return bonus;
    }
}

File: Module1_CS_OOPS/ListDemo2/Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ListDemo2
{
    internal class Employee : IComparable<Employee>
    {
        int ecode;
        string ename;
        int salary;
        int deptid;
        public Employee(int ecode,string ename,int salary,int deptid)
        {
            this.ecode = ecode;
            this.ename = ename;
            this.salary = salary;
            this.deptid = deptid;
        }

        public int GetSalary()
        {
            return salary;
        }
        public int CompareTo(Employee? other)
        {
            if(this.ecode>other.ecode)
            {
                return 1;
            }
            else if ((this.ecode<other.ecode))
            {
                return -1;
            }
            else
            {
                return 0;
            }
        }

        public override string ToString()
        {
            //return $"{ecode}\t{ename}\t{salary}\t{deptid}";
            return $"Ecode:{ecode}\nEname:{ename}\nSalary:{salary}\nDeptid:{deptid}";
        }
    }

    class SortBySalary : IComparer<Employee>
    {
        public int Compare(Employee? x, Employee? y)
        {
            if(x.GetSalary()>y.GetSalary())
            {
                return 1;
            }
            else if(x.GetSalary()<y.GetSalary()) 
            {
                return -1;
            }
            else
            {
                return 0;
            }
        }
    }
}
